tools/ops/policyrename.py

```diff
@@ -0,0 +1,172 @@
+# limitations under the License.
+"""Policy output rename utility
+"""
+
+from __future__ import print_function
+
+import argparse
+import fileinput
+import logging
+import os
+import shutil
+import sys
+
+from boto3.session import Session
+from botocore.exceptions import ClientError
+from c7n.utils import parse_s3
+
+
+log = logging.getLogger("custodian.policyrename")
+
+
+class ArgumentError(Exception): pass
+
+
+def setup_parser():
+    desc = ('This utility script will preserve the history of a policy '
+            'if it is renamed.  Pass in the old policy name and new '
+            'policy name and any old policy output and logs will be '
+            'copied to the new policy name.')
+
+    parser = argparse.ArgumentParser(description=desc)
+    parser.add_argument('-s', '--output-dir', required=True,
+                        help="Directory or S3 URL For Policy Output")
+    parser.add_argument("old", help="Old policy name")
+    parser.add_argument("new", help="New policy name")
+
+    return parser
+
+
+def s3_rename(output_dir, old, new):
+    # move the old data into the new area
+    session = Session()
+    client = session.client('s3')
+    s3 = session.resource('s3')
+    s3_path, bucket, key_prefix = parse_s3(output_dir)
+    
+    # Ensure bucket exists
+    try:
+        client.head_bucket(Bucket=bucket)
+    except ClientError:
+        raise ArgumentError('S3 bucket {} does not exist.'.format(bucket))
+
+    try:
+        log.info(
+            'Retrieving list of S3 objects to rename in bucket "{}"'.format(
+                bucket
+            )
+        )
+        to_rename = client.list_objects(Bucket=bucket, Prefix=old + '/')
+    except ClientError as e:
+        log.error(e.message)
+
+    if to_rename is None or to_rename.get('Contents') is None:
+        raise ArgumentError('Key {} does not exist in bucket {}'.format(
+                old, bucket))
+
+    # Loop through the old objects copying and deleting
+    for obj in to_rename.get('Contents'):
+        old_key = obj.get('Key')
+        new_key = new + old_key[len(old):]
+
+        # check that we haven't already run and have existing data
+        # in the new key
+        new_obj = s3.Object(bucket, new_key)
+        try:
+            new_obj.load()
+            log.info('Skipping existing output in new location: {}'.format(
+                new_obj.key))
+        except ClientError as e:
+            response_code = e.response.get('Error').get('Code')
+            if response_code == '404':
+                new_obj.copy_from(
+                    CopySource={'Bucket': bucket, 'Key': old_key}
+                )
+                log.debug('Renamed "{}" to "{}"'.format(old_key, new_key))
+        # Either way, we delete the old object
+        s3.Object(bucket, old_key).delete()
+        log.debug('Deleted "{}"'.format(old_key))
+
+
+def local_rename(output_dir, old, new):
+    """ Rename a local filesystem directory to a new name.
+    
```

Comment:
fwiw, i don't really think we need this.. local output doesn't do date segmentation, its flat, so multiple runs overwrite in place unless a user specifies their own prefix, but even then its user defined semantics that are simple as a `mv` or require interpretation (merge resources.json?). i'm okay with leaving this off. General recommend to users is to use s3 for anything considered production or valuable.